hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/SecureBulkLoadManager.java

```diff
@@ -278,6 +278,19 @@ public Map<byte[], List<Path>> run() {
               if (!fs.exists(stageFamily)) {
                 fs.mkdirs(stageFamily);
                 fs.setPermission(stageFamily, PERM_ALL_ACCESS);
+                if (
+                  conf.getBoolean(HConstants.BULKLOAD_COPYFILE_STORAGE_POLICY_ENABLED_KEY,
+                    HConstants.BULKLOAD_COPYFILE_STORAGE_POLICY_ENABLED_DEFAULT)
+                    && (!FSUtils.isSameHdfs(conf, new Path(el.getSecond()).getFileSystem(conf), fs)
```

Comment:
In which scenario request.getCopyFile() will be true while FSUtils.isSameHdfs will be true? When we are loading within same HDFS? 

Does this check handle two case: 
- Case 1: cross HDFS bulkload?
- Case 2: Within same HDFS bulkload but copyfile?

Description says we are handling only case 1 right?

Reply:
I'm sorry, my description did indeed omit the second scenario, for the second scenario (Within the same HDFS bulkload but copyfile), if the tiered storage policy is not configured when generating the HFile during the first phase of the bulkload, it can still achieve the purpose of tiered storage in the second phase by specifying -Dalways.copy.files=true, this scenario provides users with an additional option.